src/contexts/NavigationProvider.tsx

```diff
@@ -0,0 +1,109 @@
+import {
+  createContext,
+  useReducer,
+  useLayoutEffect,
+  type PropsWithChildren,
+  type Dispatch
+} from 'react'
+import { NavigationWrapper } from '@/components/NavigationWrapper'
+import type { ContentState, NavigationState, NavigationAction } from '@/types'
+import { NavigationActionType } from '@/types'
+
+import { useHistory } from '@/hooks'
+
+export const NavigationContext = createContext<{
+  state: NavigationState
+  dispatch: Dispatch<NavigationAction>
+}>({ state: {} as NavigationState, dispatch: () => {} }) // eslint-disable-line @typescript-eslint/consistent-type-assertions
+
+export const NavigationProvider = ({ children, init }: PropsWithChildren<{ init: () => NavigationState }>): JSX.Element => {
+  const [state, dispatch] = useReducer(navigationReducer, null, init)
+  const historyState = useHistory()
+
+  useLayoutEffect(() => {
+    // Create history state for initial content
+    if (historyState === null) {
+      // TODO: Make router/location hook
+      const currentView = window.location.pathname !== '/'
+        ? window.location.pathname[1].toUpperCase() + window.location.pathname.slice(2)
+        : 'Init'
+      const currentProps = Object.fromEntries(new URLSearchParams(window.location.search))
+      const id = currentProps.id || 'init' // eslint-disable-line @typescript-eslint/strict-boolean-expressions
```

Comment:
This eslint-disable can be removed **later**, when merged with add-basic-editor-and-yjs branch where I have disabled this stupid rule :)

Reply:
Yea, i figured this was temporary until we merge :slightly_smiling_face: